sdk/keyvault/keyvault/src/az_keyvault_client.c

```diff
@@ -127,7 +127,7 @@ AZ_NODISCARD az_result az_keyvault_keys_client_init(
             {
               ._internal = {
                 .process = az_http_pipeline_policy_transport,
-                .p_options= &self->_internal.options._internal.http_transport_options,
+                .p_options= &options->_internal.http_transport_options,
```

Comment:
It was working anyway right? Since we set the address, and we are also assigning a copy of options with `.options = *options,`  (and probably that is assigned before `.pipeline` then that's why it was working anyway  right?

I am OK with this change, make sense, but I got curious about why it works fine even without it

Reply:
It doesn't work without it.  It ends up calling a random address.  TBH, I'm not sure how this ever worked